.github/workflows/testing-and-coverage.yml

```diff
@@ -35,3 +35,5 @@ jobs:
         python -m pytest tests --cov=genetic_testing --cov-report=xml
     - name: Upload coverage report to codecov
       uses: codecov/codecov-action@v3
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
How does this get set/updated? Maybe add a comment in case for a future maintainer

Reply:
Good catch. I was referring to `Codecov` repository to add a comment and I found that for public repositories token isn't needed - More details [here](https://github.com/codecov/codecov-action#usage:~:text=For%20public%20repositories%2C%20no%20token%20is%20needed). So, I have removed this token.